mindsdb/integrations/handlers/chromadb_handler/chromadb_handler.py

```diff
@@ -122,144 +93,88 @@ def check_connection(self):
-                            f"where clause parameter {arg.args[0].parts[-1]} is not supported, "
-                            f"only 'meta_data_filter' and 'search_query' are supported"
-                        )
-                else:
-                    raise NotImplementedError(
-                        f"Unsupported where clause {arg.op} operator, only '=' is supported"
-                    )
-
-        elif query.where.op == "=":
-            # requires separate handling as the where clause is not a list when single condition
-            if query.where.args[0].parts[-1] == "meta_data_filter":
-                # filters on metadata
-                where["meta_data_filter"] = get_metadata_filter(
-                    query.where.args[1].value
-                )
-
-            elif query.where.args[0].parts[-1] == "search_query":
-                # extract the search query
-                where["search_query"] = query.where.args[1].value
-            else:
-                raise NotImplementedError(
-                    f"where clause parameter {query.where.args[0].parts[-1]} is not supported, "
-                    f"only 'meta_data_filter' and 'search_query' are supported"
-                )
-
-        else:
-            raise NotImplementedError(
-                f"Unsupported where clause {query.where.op} operator, only '=' and 'and' is supported"
+        # if count is used, return the count of the collection
+        if not isinstance(query.targets[0].args[0], Star):
+            # if count is not using '*' argument, raise error
+            return Response(
+                resp_type=RESPONSE_TYPE.ERROR,
+                error_message="Count only supports '*' argument",
             )
 
-        return where
+        collection_data = self._collection.count()
+        return pd.DataFrame(columns=["count"], data=[[collection_data]])
 
-    def query_collection(self, query: ASTNode) -> Response:
+    def similarity_search(self, query: ASTNode) -> Response:
         """
-        Run a select query on the ChromaDB database, filter collection (if where clause)
-        and return result. NB Limit will define the number of documents returned, not the number of rows.
+        Run a query on a chroma database
         """
 
         collection_name = query.from_table.parts[-1]
         self._collection = self._client.get_collection(collection_name)
 
         if hasattr(query.targets[0], "op") and query.targets[0].op == "count":
+            # get count of embeddings in a collection
 
-            # if count is used, return the count of the collection
-            if not isinstance(query.targets[0].args[0], Star):
-                # if count is not using '*' argument, raise error
-                return Response(
-                    resp_type=RESPONSE_TYPE.ERROR,
-                    error_message="Count only supports '*' argument",
-                )
+            return Response(resp_type=RESPONSE_TYPE.TABLE, data_frame=self.count(query))
```

Comment:
Shall we add one branch for the `select count` query in the routing and parsing logic so it does not appear within the normal `select` function (since the logics look quite different)?